### ISSUE TITLE
Allow writing code notes while testing compatibility

### ISSUE THREAD

**Issue by Alfex3** [RetroAchievements/RAIntegration#538]
While testing compatibility, it's not possible to add code notes. This makes it harder to add region support, since you have to either link and quickly unlink the ROM or leave it linked, leaving the risk of someone creating a ticket that the developer has to close because it was the wrong ROM.

**Comment by Jamiras** [comment]
This isn't the first time this has been requested, but I'm hesitant to do it as if it turns out to be incompatible, you now have a bunch of notes that don't apply to any of the linked ROMs. I'd much rather have a system where you could create local code notes and then flush them to the server when the ROM gets linked.

I've also had requests to be able to create local achievements in compatibility testing mode, which is something I plan to add (at some point) as nothing can be committed to the server until the ROM is linked.

The bottom line is compatibility _testing_ mode was added for testing, not implementing. The intent was to prevent people from linking ROMs to see if they were compatible. The thought was that if it's compatible enough, you could link it _and then_ add compatibility.

**Comment by Blazekickn** [comment]
the rom is incompatible thats why you would want to make notes without having other people able to play the game with it

**Comment by Jamiras** [comment]
> I've also had requests to be able to create local achievements in compatibility testing mode, which is something I plan to add (at some point) as nothing can be committed to the server until the ROM is linked.

You can already create local achievements in compatibility testing mode.

The following lists identifies everything that is affected:
1) User unlocks are not downloaded (all core achievements are active even if the player has earned them on the associated game)
2) Achievement unlocks are not submitted
3) Leaderboard entries are not submitted
4) User cannot report Broken Achievements
5) Rich Presence says "Testing Compatibility" when one of the toolkit dialogs is open
6) User cannot add/modify code notes
7) User cannot publish local achievements or commit unofficial/core achievements.

Having put some thought behind this, the preferred way to implement compatibility would be to change the achievements before linking the invalid ROM. As such, [7] should be allowed in compatibility mode, and if that's allowed, I see no reason not to also allow [6].

I could argue that it's still undesirable to allow adding notes to a complete set until the compatibility is guaranteed, but there's nothing that prevents a user from loading up the original ROM and adding those same notes. It's just more inconvenient.

I'm going to change it to allow [6] and [7]. [1]-[5] will remain unchanged. It'll be the dev's responsibility to make sure whatever compatibility they add doesn't break the existing published set.
